Maven also-make not working for some Quarkus dependencies

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

dist/quarkus

### Describe the bug

I want to use Maven's also-build feature to build just the Keycloak Quarkus distribution. 

I found that sometimes not all dependencies where built/updated (I was missing some local code changes), and when trying a clean build with an empty Maven cache, I saw it failing. 

Similar when trying to build just the docs.


### Version

main

### Expected behavior

Build should complete

### Actual behavior

```
[INFO] Keycloak Admin UI .................................. SUCCESS [  0.185 s]
[INFO] Keycloak Administration UI ......................... SUCCESS [  0.020 s]
[INFO] Admin UI REST extensions ........................... SUCCESS [  0.061 s]
[INFO] Keycloak Quarkus Server Extension .................. FAILURE [  4.856 s]
[INFO] Keycloak Integration ............................... SKIPPED
[INFO] Keycloak Client CLI ................................ SKIPPED
[INFO] Keycloak Client Registration CLI ................... SKIPPED
[INFO] Keycloak Admin CLI ................................. SKIPPED
[INFO] Keycloak Client CLI Distribution ................... SKIPPED
[INFO] Keycloak Quarkus Server App ........................ SKIPPED
[INFO] Keycloak Quarkus Server Distribution ............... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10.814 s
[INFO] Finished at: 2023-09-06T17:35:00+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-extension-maven-plugin:3.2.5.Final:extension-descriptor (default) on project keycloak-quarkus-server: Deployment artifact org.keycloak:keycloak-quarkus-server-deployment::jar:999.0.0-SNAPSHOT is missing the following dependencies from its configuration: org.keycloak:keycloak-quarkus-server::jar, io.quarkus:quarkus-vertx-http-deployment::jar, io.quarkus:quarkus-mutiny-deployment::jar, io.quarkus:quarkus-smallrye-context-propagation-deployment::jar, io.quarkus:quarkus-vertx-deployment::jar, io.quarkus:quarkus-arc-deployment::jar, io.quarkus:quarkus-netty-deployment::jar, io.quarkus:quarkus-reactive-routes-deployment::jar, io.quarkus:quarkus-jackson-deployment::jar, io.quarkus:quarkus-resteasy-deployment::jar, io.quarkus:quarkus-resteasy-server-common-deployment::jar, io.quarkus:quarkus-resteasy-common-deployment::jar, io.quarkus:quarkus-resteasy-jackson-deployment::jar, io.quarkus:quarkus-hibernate-orm-deployment::jar, io.quarkus:quarkus-narayana-jta-deployment::jar, io.quarkus:quarkus-caffeine-deployment::jar, io.quarkus:quarkus-agroal-deployment::jar, io.quarkus:quarkus-datasource-deployment::jar, io.quarkus:quarkus-jdbc-h2-deployment::jar, io.quarkus:quarkus-jdbc-postgresql-deployment::jar, io.quarkus:quarkus-jdbc-mariadb-deployment::jar, io.quarkus:quarkus-jdbc-mysql-deployment::jar, io.quarkus:quarkus-jdbc-mssql-deployment::jar, io.quarkus:quarkus-jdbc-oracle-deployment::jar, io.quarkus:quarkus-core-deployment::jar, io.quarkus:quarkus-smallrye-health-deployment::jar, io.quarkus:quarkus-jsonp-deployment::jar, io.quarkus:quarkus-micrometer-deployment::jar, io.quarkus:quarkus-micrometer-registry-prometheus-deployment::jar, io.quarkus:quarkus-logging-json-deployment::jar, io.quarkus:quarkus-logging-gelf-deployment::jar -> [Help 1]
```

### How to Reproduce?

```
rm -rf ~/.m2/repository/org/keycloak
mvn -am -pl quarkus/dist clean install -DskipTests
```

### Anything else?

I'm preparing a PR

--- a/docs/maven-plugin/pom.xml
+++ b/docs/maven-plugin/pom.xml
@@ -56,7 +56,7 @@
         </dependency>
         <dependency>
             <groupId>org.keycloak</groupId>
-            <artifactId>keycloak-quarkus-server</artifactId>
+            <artifactId>keycloak-quarkus-server-deployment</artifactId>
         </dependency>
         <dependency>
             <groupId>org.freemarker</groupId>

